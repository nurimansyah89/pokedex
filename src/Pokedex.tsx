import * as React from 'react';
import { Provider } from 'react-redux';
import Store from './redux/store';
import Router from './Router';
import '@fortawesome/fontawesome-free/css/all.css';
import './styles/pokedex.scss';

const Pokedex = () => (
  <Provider store={Store}>
    <Router />
  </Provider>
);
export default Pokedex;
