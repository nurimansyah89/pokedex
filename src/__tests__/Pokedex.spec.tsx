import * as React from 'react';
import renderer from 'react-test-renderer';
import Pokedex from '../Pokedex';
import Choices from 'choices.js';

jest.genMockFromModule('choices.js');
jest.mock('choices.js');

describe('Pokedex Test', () => {
  const originalScrollTo = window.scrollTo;

  beforeEach(() => {
    const _: jest.Mock<Choices> = new Choices(document.createElement('select')) as any;
  });

  it('Should render without crash', () => {
    window.scrollTo = () => null;
    const tree = renderer.create(<Pokedex />).toJSON();
    expect(tree).toMatchSnapshot();
    window.scrollTo = originalScrollTo;
  });
});
