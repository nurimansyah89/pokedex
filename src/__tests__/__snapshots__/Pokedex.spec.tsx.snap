// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokedex Test Should render without crash 1`] = `
Array [
  <div
    className="Wrapper"
  >
    <div
      className="container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1>
          Pokédex
        </h1>
      </a>
    </div>
    <div
      className="FormBackground"
    >
      <div
        className="container FormContainer"
      >
        <section>
          <form
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="field"
            >
              <label
                htmlFor="query"
              >
                Name or Number
              </label>
              <div
                className="field-group"
              >
                <select
                  id="query"
                  name="query"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Search Pokémon"
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <option
                    placeholder="Search Pokémon"
                  >
                    Search Pokémon
                  </option>
                  <option>
                    Bulbasaur
                  </option>
                  <option>
                    Charmender
                  </option>
                  <option>
                    Squirtle
                  </option>
                </select>
                <button
                  className="button button-orange"
                  type="submit"
                >
                  <span
                    className="fas fa-search"
                  />
                </button>
              </div>
            </div>
            <div
              className="field"
            >
              <p>
                Use the Advanced Search to explore Pokémon by type, weakness, Ability, and more!
              </p>
            </div>
          </form>
        </section>
        <section>
          <p
            className="notification notification-green"
            style={
              Object {
                "fontSize": "1.25rem",
              }
            }
          >
            Search for a Pokémon by name or using its National Pokédex number.
          </p>
        </section>
      </div>
    </div>
    <div
      className="AdvanceSearchContainer"
    >
      <div
        className="container Wrapper"
      >
        <div
          className="Selected"
        >
          <ul>
            <li>
              <span
                className="SelectedType"
              >
                Type:
              </span>
              <span
                className="SelectedValue"
              >
                2 Selected
              </span>
              <button
                type="button"
              >
                <span
                  className="fas fa-times"
                />
              </button>
            </li>
          </ul>
        </div>
        <button
          className="ToggleButton"
          onClick={[Function]}
          type="button"
        >
          Show Advance Search
           
          <span
            className="fa-stack"
          >
            <i
              className="fas fa-circle fa-stack-1x"
            />
            <i
              className="fas fa-angle-down fa-stack-1x"
            />
          </span>
        </button>
        <div
          className="ContentWrapper"
        >
          <div
            className="Filter"
          >
            <section>
              <div
                className="FilterType"
              >
                <h2>
                  Type & Weakness
                </h2>
                <span>
                  <strong>
                    T
                  </strong>
                   = Type
                </span>
                <span>
                  <strong>
                    W
                  </strong>
                   = Weakness
                </span>
              </div>
              <ul
                className="TypeList"
              >
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#729f3f",
                          "color": "white",
                        }
                      }
                    >
                      Bug
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#53a4cf",
                          "color": "white",
                        }
                      }
                    >
                      Dragon
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#fdb9e9",
                          "color": undefined,
                        }
                      }
                    >
                      Fairy
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#fd7d24",
                          "color": "white",
                        }
                      }
                    >
                      Fire
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#7b62a3",
                          "color": "white",
                        }
                      }
                    >
                      Ghost
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#7b62a3",
                          "color": "white",
                        }
                      }
                    >
                      Ground
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#7b62a3",
                          "color": "white",
                        }
                      }
                    >
                      Normal
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#7b62a3",
                          "color": "white",
                        }
                      }
                    >
                      Physic
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#7b62a3",
                          "color": "white",
                        }
                      }
                    >
                      Steel
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#7b62a3",
                          "color": "white",
                        }
                      }
                    >
                      Dark
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#7b62a3",
                          "color": "white",
                        }
                      }
                    >
                      Electric
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="TypeContainer"
                  >
                    <div
                      className="TypeName"
                      style={
                        Object {
                          "background": "#7b62a3",
                          "color": "white",
                        }
                      }
                    >
                      Fighting
                    </div>
                    <div
                      className="TypeAction"
                    >
                      <button
                        type="button"
                      >
                        T
                      </button>
                      <button
                        type="button"
                      >
                        W
                      </button>
                    </div>
                  </div>
                </li>
              </ul>
            </section>
            <section>
              <div
                className="FilterAbility"
              >
                <h2>
                  Ability
                </h2>
                <div
                  className="dropdown dropdown__pokedrop"
                >
                  <select
                    id="ability"
                    name=""
                  >
                    <option
                      value=""
                    >
                      All
                    </option>
                  </select>
                </div>
              </div>
              <div
                className="FilterHeight"
              >
                <h2>
                  Height
                </h2>
                <ul>
                  <li
                    className="FilterBox HeightSmall"
                  >
                    <button
                      type="button"
                    />
                  </li>
                  <li
                    className="FilterBox HeightMedium"
                  >
                    <button
                      type="button"
                    />
                  </li>
                  <li
                    className="FilterBox HeightLarge"
                  >
                    <button
                      type="button"
                    />
                  </li>
                </ul>
              </div>
              <div
                className="FilterWeight"
              >
                <h2>
                  Weight
                </h2>
                <ul>
                  <li
                    className="FilterBox WeightSmall"
                  >
                    <button
                      type="button"
                    />
                  </li>
                  <li
                    className="FilterBox WeightMedium"
                  >
                    <button
                      type="button"
                    />
                  </li>
                  <li
                    className="FilterBox WeightLarge"
                  >
                    <button
                      type="button"
                    />
                  </li>
                </ul>
              </div>
            </section>
          </div>
          <div
            className="Range"
          >
            <h2>
              Number Range
            </h2>
            <div
              className="RangeInput"
            >
              <input
                className="input"
                id=""
                max="1000"
                min="1"
                name=""
                placeholder="1"
                type="number"
              />
              <span>
                -
              </span>
              <input
                className="input"
                id=""
                max="1000"
                min="1"
                name=""
                placeholder="1000"
                type="number"
              />
            </div>
          </div>
          <div
            className="Action"
          >
            <input
              className="button button-gray"
              type="reset"
              value="Reset"
            />
            <button
              className="button button-orange"
              type="submit"
            >
              <span
                className="fas fa-search"
              />
               Search
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container Wrapper"
  >
    <div
      className="dropdown dropdown__pokedrop color-white"
    >
      <select
        id="sortDropdown"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Sort results by...
        </option>
        <option
          value="POKEMON_SORT_BY_LOWEST_NUMBER"
        >
          Lowest Number (First)
        </option>
        <option
          value="POKEMON_SORT_BY_HIGHEST_NUMBER"
        >
          Highest Number (First)
        </option>
        <option
          value="POKEMON_SORT_BY_A_Z"
        >
          A-Z
        </option>
        <option
          value="POKEMON_SORT_BY_Z_A"
        >
          Z-A
        </option>
      </select>
    </div>
  </div>,
  <div
    className="container Wrapper"
  >
    <ul
      className="ListWrapper"
    />
    <div
      className="align-center loader"
    >
      <span
        className="fas fa-spin fa-spinner fa-2x"
      />
    </div>
  </div>,
  <footer
    className="Wrapper"
  >
    <div
      className="container"
    >
      <h2>
        Pokédex - A Pokémon Encyclopedia
      </h2>
      <p
        className="Author"
      >
        © 2019, Nurimansyah Rifwan.
      </p>
      <p
        className="Copyright"
      >
        <small>
          Please note that all of the assets are trademarks and copyrights of their respective owners. Though, I'm not affiliated with 
          <strong>
            Nintendo, The Pokémon Company Creatures Inc.
          </strong>
           or 
          <strong>
            Game Freak
          </strong>
          .
        </small>
      </p>
    </div>
  </footer>,
]
`;
